src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -99,20 +105,46 @@ namespace {
 NetworkInfo sNetworks[kMaxNetworks];
 } // namespace
 
-// TODO: We cannot get length of operational dataset here, should be fixed after we have byte string type.
-// Note: The codegen for OCTET_STRING will be implemented later since no other clusters are using OCTET_STRING.
 EmberAfNetworkCommissioningError OnAddThreadNetworkCommandCallbackInternal(app::Command *, EndpointId,
-                                                                           const uint8_t * operationalDataset, uint64_t breadcrumb,
+                                                                           const uint8_t * operationalDataset,
+                                                                           size_t operationalDatasetLen, uint64_t breadcrumb,
                                                                            uint32_t timeoutMs)
 {
-    // TODO: Implement AddThreadNetworkCommand after OCTET_STRING is supported.
-    return EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_NETWORK_NOT_FOUND;
+    EmberAfNetworkCommissioningError err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_BOUNDS_EXCEEDED;
+
+    for (size_t i = 0; i < kMaxNetworks; i++)
+    {
+        if (sNetworks[i].mNetworkType == NetworkType::kUndefined)
+        {
+            VerifyOrExit(operationalDatasetLen <= kMaxThreadDatasetLen, err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_OUT_OF_RANGE);
+            memcpy(sNetworks[i].mData.mThread.mDataset, operationalDataset, operationalDatasetLen);
+            // We have verified that the operationalDatasetLen is less than kMaxThreadDatasetLen.
+            sNetworks[i].mData.mThread.mDatasetLen = static_cast<uint8_t>(operationalDatasetLen);
+
+            // A 64bit id for thread networks, currently, we are missing some API for getting the thread network extpanid from the
+            // dataset tlv.
+            // TODO: Set thread network id to real ext panid from dataset tlv.
+            memcpy(sNetworks[i].mNetworkID, kTemporaryThreadNetworkId, sizeof(kTemporaryThreadNetworkId));
```

Comment:
Can we static_assert `sizeof(sNetworks[i].mNetworkID) >= sizeof(kTemporaryThreadNetworkId)` please?

Reply:
This also is not addressed yet?